src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -96,6 +112,62 @@ ov::PartialShape program_node::get_output_pshape(size_t idx) const {
     return get_output_layout(idx).get_partial_shape();
 }
 
+std::vector<layout> program_node::get_shape_info_input_layouts() const {
+    std::vector<layout> res;
+    for (size_t i = 0; i < get_dependencies().size(); i++) {
+        const auto& d = get_dependency_with_port(i);
+        res.push_back(d.first->get_output_layout(false, d.second));
+    }
+
+    return res;
+}
+
+std::map<size_t, size_t> program_node::get_input_port_to_shape_info_offset_map() const {
+    std::map<size_t, size_t> res;
+    size_t offset = 0;
+    const auto& deps = get_shape_info_input_layouts();
+    for (size_t i = 0; i < deps.size(); i++) {
+        res[i] = offset;
+        offset += get_shape_data_size(deps[i]);
+    }
+
+    return res;
+}
+
+std::map<size_t, size_t> program_node::get_output_port_to_shape_info_offset_map() const {
+    std::map<size_t, size_t> res;
+    size_t offset = get_total_shape_info_input_size();
+    for (size_t i = 0; i < output_layouts.size(); i++) {
+        res[i] = offset;
+        offset += get_shape_data_size(output_layouts[i]);
+    }
+
+    return res;
+}
+
+size_t program_node::get_total_shape_info_input_size() const {
+    size_t offset = 0;
+    const auto& deps = get_shape_info_input_layouts();
+    for (size_t i = 0; i < deps.size(); i++) {
+        offset += get_shape_data_size(deps[i]);
+    }
+
+    return offset;
+}
+
+size_t program_node::get_total_shape_info_output_size() const {
+    size_t offset = 0;
+    for (size_t i = 0; i < output_layouts.size(); i++) {
+        offset += get_shape_data_size(output_layouts[i]);
+    }
+
+    return offset;
+}
+
+size_t program_node::get_total_shape_info_size() const {
```

Comment:
Shouldn't this func be called in `primitive_inst::primitive_inst`?

Reply:
Yes, it should. Thanks